comp/core/secrets/secretsimpl/walker.go

```diff
@@ -0,0 +1,124 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package secretsimpl
+
+import (
+	"github.com/DataDog/datadog-agent/comp/core/secrets"
+)
+
+type resolverCallback func([]string, string) (string, error)
+
+type walker struct {
+	// resolver is called to fetch the value of a handle
+	resolver resolverCallback
+	// notifier is called each time a key from the YAML is updated. This is used by ResolveWithCallback.
+	//
+	// When a slice is updated, this will be called once with the final slice content.
+	notifier secrets.ResolveCallback
+
+	// notificationPending is true when a notification needs to be send. See documentation for ResolveCallback for
+	// more information.
+	notificationPending bool
+}
+
+func (w *walker) notify(yamlPath []string, value any) {
+	if w.notifier != nil {
+		if !w.notifier(yamlPath, value) {
+			// notification was refuse, will retry from the parent type.
```

Comment:
The code is not entirely clear which is the parent type. All variables are called `yamlPath`, and `data`. Is there a way to rename some variables to make it more apparent where the notification with the parent happens?

Reply:
I renamed variables and added comment to the walker. Tell me what you think.